processing/src/main/java/org/apache/carbondata/processing/loading/iterator/CarbonOutputIteratorWrapper.java

```diff
@@ -98,14 +99,31 @@ public void closeWriter(boolean isForceClose) {
         close = true;
         return;
       }
-      loadBatch.readyRead();
-      if (loadBatch.size > 0) {
-        queue.put(loadBatch);
+      // flag to check whether all the RowBatches have been consumed
+      boolean processingFinished = false;
```

Comment:
As this class will be created for each thread as mentioned in earlier comment, i think below changes is also not required. In case of multi threaded each writer will have its own CarbonOutputIteratorWrapper object.

Reply:
As discussed with you this change is required because put is a blocking call and if consumer is slow or any exception occurs then it is a possibility that the thread will still be waiting at put method for the last RowBatch